odoo/addons/base/models/ir_mail_server.py

```diff
@@ -211,6 +251,14 @@ def connect(self, host=None, port=None, user=None, password=None, encryption=Non
             if smtp_encryption is None and tools.config.get('smtp_ssl'):
                 smtp_encryption = 'starttls' # smtp_ssl => STARTTLS as of v7
 
+            smtp_ssl_certificate = ssl_certificate or tools.config.get('smtp_ssl_certificate')
+            smtp_ssl_private_key = ssl_private_key or tools.config.get('smtp_ssl_private_key')
+            if smtp_ssl_certificate and smtp_ssl_private_key:
+                ssl_context = ssl.SSLContext()
+                ssl_context.load_cert_chain(smtp_ssl_certificate, keyfile=smtp_ssl_private_key)
```

Comment:
Linting: Couldn't we compute certificates and keyfiles once, and to the SSLContext() + load_cert_chain once, instead of quite duplicating code ? Notably for error management for example, I bet it could be useful to have calls to SSL library only once.

Reply:
For the temporary file creation it's more complicated to make a common function because we should use a context to remove the temporary file after utilization